R/potassium.R

```diff
@@ -0,0 +1,195 @@
+#' Calculate the K availability
+#' 
+#' This function calculates the K availability of a soil.
+#' 
+#' @param A_PH_CC (numeric) The acidity of the soil, measured in 0.01M CaCl2 (-)
+#' @param A_OS_GV (numeric) The organic matter content of the soil (in procent)
+#' @param A_CEC_CO (numeric) The cation exchange capacity of the soil (mmol+ per kg), analysed via Cobalt-hexamine extraction
+#' @param A_K_CC (numeric) The plant available potassium, extracted with 0.01M CaCl2 (mg per kg), 
+#' @param A_K_CEC (numeric) The occupation of the CEC with potassium (in procent)
+#' @param A_CLAY_MI (numeric) The clay content of the soil (in procent)
+#' @param B_BT_AK (character) The type of soil
+#' @param B_LU_BRP (numeric) The crop code (gewascode) from the BRP
+#' 
+#' @import data.table
+#' 
+#' @export
+calc_potassium_availability <- function(A_K_CC, A_K_CEC,A_CEC_CO, A_PH_CC, A_OS_GV, A_CLAY_MI, 
+                                        B_LU_BRP, B_BT_AK) {
+  
+  id = crop_category = soiltype.n = crop_code = soiltype = NULL
+  b = cF = kindex1 = kindex2 = A_PH_KCL = A_K_CO = NULL
+  
+  # Load in the datasets
+  crops.obic <- as.data.table(OBIC::crops.obic)
+  setkey(crops.obic, crop_code)
+  soils.obic <- as.data.table(OBIC::soils.obic)
+  setkey(soils.obic, soiltype)
+  
+  # Check inputs
+  arg.length <- max(length(A_PH_CC), length(A_OS_GV), length(A_CEC_CO), length(A_K_CEC), 
+                    length(A_K_CC), length(A_CLAY_MI), length(B_BT_AK), length(B_LU_BRP))
+  checkmate::assert_numeric(A_PH_CC, lower = 3, upper = 10, any.missing = FALSE, len = arg.length)
```

Comment:
waarden deze checks in help geven. dan weet de gebruiker wat de mogelijke ranges zijn